test/com/amazon/ion/impl/bin/IonManagedBinaryWriterTest.java

```diff
@@ -104,6 +226,54 @@ public void testFlushImmediatelyAfterIVM() throws Exception
         assertEquals("burrito", ((IonSymbol) dg.systemGet(2)).stringValue());
     }
 
+    @Test
+    public void testAutoFlush() throws Exception{
+        IonReader reader = system().newReader(source.toByteArray());
+        ByteArrayOutputStream actual = new ByteArrayOutputStream();
+        IonWriter actualWriter = IonBinaryWriterBuilder.standard().withAutoFlushEnabled(autoFlushMode.isEnabled()).build(actual);
+        while (reader.next() != null) {
+            actualWriter.writeValue(reader);
+        }
+        actualWriter.close();
+        if (lstAppendMode.isEnabled() && autoFlushMode.isEnabled()) {
+            assertEquivalentDataModel(actual, expectedOut);
+        }
+        // When auto-flush mode is disabled, the serialized data should be equivalent to the source data without flush operation.
+        assertEquivalentDataModel(actual, source);
+    }
+
+    @Test
+    public void testAutoFlush_32K() throws Exception{
+        IonReader reader = system().newReader(source_32K.toByteArray());
+        ByteArrayOutputStream actual = new ByteArrayOutputStream();
+        IonWriter actualWriter = IonBinaryWriterBuilder.standard().withAutoFlushEnabled(autoFlushMode.isEnabled()).build(actual);
+        while (reader.next() != null) {
+            actualWriter.writeValue(reader);
+        }
+        actualWriter.close();
+        if (lstAppendMode.isEnabled() && autoFlushMode.isEnabled()) {
+            assertEquivalentDataModel(actual, expectedOut_32K);
+        }
+        // When auto-flush mode is disabled, the serialized data should be equivalent to the source data without flush operation.
+        assertEquivalentDataModel(actual, source_32K);
+    }
+
+
+    @Test
+    public void testAutoFlush_67K() throws Exception{
+        IonReader reader = system().newReader(source_67K.toByteArray());
+        ByteArrayOutputStream actual = new ByteArrayOutputStream();
+        IonWriter actualWriter = IonBinaryWriterBuilder.standard().withAutoFlushEnabled(autoFlushMode.isEnabled()).build(actual);
+        while (reader.next() != null) {
+            actualWriter.writeValue(reader);
+        }
+        actualWriter.close();
+        if (lstAppendMode.isEnabled() && autoFlushMode.isEnabled()) {
+            assertEquivalentDataModel(actual, expectedOut_67K);
```

Comment:
Here you actually do want to compare the bytes, right? To ensure the auto-flush occurred.